preprocess.py

```diff
@@ -0,0 +1,153 @@
+    """
+    ud_lines: List[str] = read_file_as_lines(ud_file)
+    with open(ud_file) as infile:
+        ud_lines = infile.readlines()
+    ud_sentences: List[str] = []
+    for line in ud_lines:
+        if "# text" in line:
+            text: str = line.split(" text = ")[1]
+            sentence: str = substitute_brackets(text)
+            ud_sentences.append(sentence)
+        else:
+            continue
+    return ud_sentences
+
+
+def process_um_data(um_file: str) -> List[str]:
+    """Processes UniMorph file into list of strings for normalization.
+
+    UM data format is the word lemma, followed by a tab, followed by
+    the inflected form of the word, followed by another tab, followed by
+    fine-grained morphological information."""
+    raise NotImplementedError
+
+
+def process_ancrubadan_data(ac_file: str) -> List[str]:
+    """Processes An Crubadan file into list of strings for normalization.
+
+    An Crubadan data format is the word, followed by a space, followed by
+    the totalcount of that word in the corpora that were crawled.
+
+    Args:
+        ac_file: The ac data file to process.
+
+    Returns:
+        A list of processed words.
+    """
+    ac_lines = read_file_as_lines(ac_file)
+    ac_words: List[str] = []
+    for line in ac_lines:
+        word: str = substitute_brackets(line)
+        ac_words.append(word)
+    return ac_words
+
+
+def process_oscar_data(oscar_file: str) -> List[str]:
+    """Processes OSCAR file into list of strings for normalization.
+
+    OSCAR data format is one (or more) sentences per line. This processing
+    does not separate these into separate sentences.
+
+    Args:
+        ac_file: The ac data file to process.
+
+    Returns:
+        A list of processed strings consisting of one or more sentences.
+    """
+    oscar_lines = read_file_as_lines(oscar_file)
+    oscar_strings: List[str] = []
+    for line in oscar_lines:
+        sentence: str = substitute_brackets(line)
+        oscar_strings.append(sentence)
+    return oscar_strings
+
+
+def process_lcc_data(lcc_file: str) -> List[str]:
+    """Processes Leipzig Corpora Collection file for normalization.
+
+    LCC data format is the sentence index, followed by a tab,
+    followed by the sentence text.
+
+    Args:
+        lcc_file: The lcc data file to process.
+
+    Returns:
+        A list of processed sentences.
+    """
+    lcc_lines = read_file_as_lines(lcc_file)
+    lcc_sentences: List[str] = []
+    for line in lcc_lines:
+        text: str = line.strip().split("\t")[1]
+        sentence: str = substitute_brackets(text)
+        lcc_sentences.append(sentence)
+    return lcc_sentences
+
+
+def read_file_as_lines(filename: str) -> List[str]:
+    """Reads in filename as list of lines."""
+    with open(filename) as infile:
+        file_lines = infile.readlines()
+    return file_lines
+
+
+def substitute_brackets(string: str) -> str:
+    """Substitutes square brackets to work in Pynini."""
```

Comment:
Could you add args and returns to these docstrings?

Reply:
Done!

Reply:
Thanks!